### PR TITLE
feat(xref/ui): pre-fill form via url search params

### PR DESCRIPTION
**Pull request by sidvishnoi**

### CODE REVIEW

File: static/xref/script.js [hunk 1]
```diff
@@ -95,7 +95,9 @@ function getFormData() {
 }
 
 async function onSubmit(event) {
-  event.preventDefault();
+  if (event && typeof event.preventDefault === "function") {
```
Comment (marcoscaceres) [inline static/xref/script.js]: any particular reason for this check?

Comment (sidvishnoi) [reply static/xref/script.js]: wanted to reuse `onSubmit` below

Comment (sidvishnoi) [reply static/xref/script.js]: ...to call `onSubmit` without any event

Comment (marcoscaceres) [reply static/xref/script.js]: ah, maybe put a comment that this is not a support check.

Comment (sidvishnoi) [reply static/xref/script.js]: makes sense

File: static/xref/script.js [hunk 2]
```diff
@@ -187,6 +189,23 @@ async function ready() {
...
+        form[field].value = value;
+        break;
+      case "cite":
+      case "types":
+        value.split(",").forEach(val => form[field].select(val));
+        break;
+    }
+  }
+  if (searchParams.has("term")) {
+    onSubmit();
```
Comment (marcoscaceres) [inline static/xref/script.js]: Alternative, just pass a CustomEvent?

Comment (marcoscaceres) [reply static/xref/script.js]: prevenDefault() on custom event will be a no-op.

Comment (sidvishnoi) [reply static/xref/script.js]: how about remove the event argument altogether and handle `preventDefault` at `form.addEventListener("submit", ...)`?

Comment (marcoscaceres) [reply static/xref/script.js]: Yeah, I think just send a new CustomEvent here... then the other code above doesn't need the special check.

Comment (sidvishnoi) [reply static/xref/script.js]: how about 0457ba1?

**Pull request merged**